Clothing armour values on examine

cacogen opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

When clothing has armour and you examine it it'll give you a hyperlink to a list of armour and an estimate of its strength.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

No more is armour meta information for code divers while everyone else relies on incomplete descriptions, names and sprites.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl: cacogen
add: Space's big players are more beholden to OSH than one might've thought. Newly manufactured clothing now comes with a tag listing its protection ratings (if applicable). Deciding which armour to "borrow" from the armoury has never been easier.
/:cl:

### Todo
- [x] Hide values behind link on single line

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

cacogen committed:
Clothing armour values on examine

TheVekter commented:
I'm not opposed to this idea, especially if the exact numbers are obfuscated, but Examine is getting pretty bloated on some items.

Tlaltecuhtli commented on code/modules/clothing/clothing.dm:
```diff
@@ -140,6 +140,52 @@
+			. = "High"
+		if (60 to 80)
+			. = "Very high"
+		if (80 to 100)
```
shouldnt it be 81 etc

cacogen replied:
i cant mat

cacogen committed:
relearn year 1 maths

cacogen commented:
Examine is getting pretty bloated on some items. Originally it was a sentence that could take a couple of lines but I found a list cleaner and easier to parse. To clarify armour values of 0 don't receive a message.

spookydonut commented:
This will make examining any clothing spit out massive text walls, self documenting help pages like bays/tgmc's codex would be better for this imo.

Tlaltecuhtli commented:
just put this on the science goggles or fire the extra info if you have your clothes in hand while examinating

cacogen commented:
I think you're overestimating how often you examine clothing and how long the armour lists of the average clothes are

Tlaltecuhtli commented:
problem is when like i m trying to shift click some dude running in the hall but instead i shift click a pair of pants making me lag for 1s and maybe miss the "joe is traitor" message

Time-Green commented:
Maybe make armor put a link in the chat that, once clicked, throws the armor values in the chat

cacogen commented:
Yeah, that's what I'll do

necromanceranne commented:
Might want to add stuff like details about the items flags, like fireproofing, acidproofing, magic immunity and what they mean.

anconfuzedrock commented:
Bad post:
combat related info like armour values could only appear if examined on a non help intent to reduce bloat

Anonmare commented:
I feel like we need a seperate verb for thorough examinations like this.

optimumtact commented:
make it a popup on hover over an anchor link in goonchat. Although mind you, with the weird chat scrolling bugs that might be kind of annoying.

PKPenguin321 commented:
We should have a thing where if you examine an item twice in a row you examine it more closely

ArcaneMusic commented:
Alternatively, you could lock seeing armor values behind the sechud or the science goggles.

cacogen committed:
it's a link now

cacogen committed:
PROTECTION CLASSES (I-V)

rating system is worthless if you don't know the range

cacogen committed:
nothing i do seems complex enough to autodoc

but here is a good attempt

cacogen commented:
>Also, some of the armor values are immensely deceptive. Fire doesn't necessarily mean protection from fire. It means that armors ability to withstand fire. It might be worth splitting the interface into item durability protections and armor.

I'm not really clear on how armor values work, probably because they've been buried for so long. If you tell me which values pertain to the armor's durability and which pertain to the wearer's protection I'll split the values into those categories.

cacogen commented:
>Also, some of the armor values are immensely deceptive. Fire doesn't necessarily mean protection from fire. It means that armors ability to withstand fire. It might be worth splitting the interface into item durability protections and armor.

@necromanceranne 
I'm not really clear on how armor values work, probably because they've been buried for so long. If you tell me which values pertain to the armor's durability and which pertain to the wearer's protection I'll split the values into those categories.

@ArcaneMusic 
>Alternatively, you could lock seeing armor values behind the sechud or the science goggles.

I think this is too important to gate behind a HUD and should be readily available information.

SkeletalElite commented:
> > Also, some of the armor values are immensely deceptive. Fire doesn't necessarily mean protection from fire. It means that armors ability to withstand fire. It might be worth splitting the interface into item durability protections and armor.

Fire and Acid armor values are about protecting the item. When it comes to protecting yourself from fire, you need to be insulated from the fire.

cacogen commented:
All right, I'll add those two values to a separate list under a separate subheading

cacogen committed:
armor classes now split between armor and durability

ExcessiveUseOfCobblestone commented on code/modules/clothing/clothing.dm:
```diff
@@ -140,6 +143,70 @@
+			. = "V"
+		if (100 to INFINITY)
+			. = "∞"
+	return .
```
this doesn't need to be a proc, just edit durability amount after 188 to convert itself into level then just print out `[durability_amount]`.

necromanceranne replied:
You should have that from 1-19, 20-29 and so on, because as it stands the Elite Hardsuit and Deathsquard Hardsuit are in the same category when they are nowhere remotely the same level of protection (the Deathsquad hardsuit is exactly 80 and is the highest of any armor)

cacogen replied:
ok

cacogen committed:
Update clothing.dm

cacogen committed:
i program my home computer

cacogen committed:
oh i'm sayin it

cacogen committed:
underline instead of blank line

too many lines makes it hard to compare values

ExcessiveUseOfCobblestone commented on code/modules/clothing/clothing.dm:
```diff
@@ -140,6 +143,78 @@
+
+		to_chat(usr, "[readout.Join()]")
+
+/obj/item/clothing/proc/number_to_level(number)
```
add doc for this and should be good to go

ExcessiveUseOfCobblestone replied:
ping on discord

cacogen committed:
autodoc i hope it's good

ExcessiveUseOfCobblestone commented:
https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md#documenting-a-proc

cacogen committed:
autodocs proc properly

- changes vague number var to armor
- renames proc to be more descriptive
- makes it so armor value rounds to nearest 10 instead of implicitly rounding down to 1 sig fig
- changes armor readout formatting to hopefully look better
- simplifies code by dividing the armor value by 10 before converting to roman numerals
